.github/workflows/build-rpm.yml

```diff
@@ -22,28 +22,29 @@ jobs:
       uses: actions/checkout@v4
 
     - name: Create workspace directory
-      run: mkdir -vp workspace/{RPMS,SRPMS,SOURCES,SPECS}
+      run: mkdir -vp workspace/{RPMS,SRPMS}
 
-    - name: Copy spec file
-      run: cp ${{ matrix.el_flavor }}${{ matrix.el_version }}/*.spec workspace/SPECS/
-
-    - name: Copy source files
-      run: cp -r ${{ matrix.el_flavor }}${{ matrix.el_version }}/SOURCES/* workspace/SOURCES/
+    - name: Prepare build context
+      run: |
+        mkdir -vp build-context/{SOURCES,SPECS}
+        cp ${{ matrix.el_flavor }}${{ matrix.el_version }}/*.spec build-context/SPECS/
+        cp -r ${{ matrix.el_flavor }}${{ matrix.el_version }}/SOURCES/* build-context/SOURCES/
+        cp build-rpms.sh build-context/
 
     - name: Build Docker image
       run: |
         docker build \
                --build-arg EL_VERSION=${{ matrix.el_version }} \
                --build-arg SPEC_FILE=libvirt.spec \
                -f Dockerfile \
-               -t rpm-builder:${{ matrix.el_flavor }}${{ matrix.el_version }} workspace
+               -t rpm-builder:${{ matrix.el_flavor }}${{ matrix.el_version }} build-context
 
     - name: Run builder container
       run: |
         docker run --rm -t \
```

Comment:
Don't remove the container. We need to make sure the container state is not deleted so we can pull the artifacts from the container